src/core/stream/orchestrator/stream_orchestrator.ts

```diff
@@ -421,6 +423,30 @@ export default function StreamOrchestrator(
         } else if (type === "active-stream") {
           // current Stream is active, destroy next Stream if created
           destroyNextStreams$.next();
+        } else if (
+          evt.type === "adaptationChange" &&
+          evt.value.type === "audio" &&
+          !evt.value.isFirstAdaptation
+          ) {
+          isReadyToAdaptAudioSwitchingStategy = true;
+        } else if (
+          evt.type === "added-segment" &&
+          evt.value.content.adaptation.type === "audio" &&
+          isReadyToAdaptAudioSwitchingStategy
+          ) {
+           const { audioTrackSwitchingMode } = options;
+           isReadyToAdaptAudioSwitchingStategy = false;
+          if (audioTrackSwitchingMode === "reload") {
```

Comment:
And I also just noticed. Why do we wait to have pushed a segment before reloading?

This will just be a waste of a request (without considering client-side caching). We should reload right when the adaptation is switched.